x-pack/plugins/maps/public/connected_components/layer_panel/view.js

```diff
@@ -70,7 +75,9 @@ export class LayerPanel extends React.Component {
 
     const immutableSourceProps = await this.props.selectedLayer.getImmutableSourceProperties();
     if (this._isMounted) {
-      this.setState({ immutableSourceProps });
+      if (!_.isEqual(this.state.immutableSourceProps, immutableSourceProps)) {
```

Comment:
Why was this change needed? `immutableSourceProps` should not be updated. If that is the case then the source props is not immutable.

Reply:
Depends if we want to show "mutable" source-properties here or not. 

![image](https://user-images.githubusercontent.com/1833023/86049029-6cc7b000-ba1f-11ea-8d72-9c02623fdf81.png)

I'd just rename the method to `getSourceProperties`. Not 100% sure if "immutable"-part is relevant.
No strong preference either way.

Reply:
As discussed, lets not show mutable properties since those are shown in the edit forms below. This section is just to show stuff the user can not edit but may need to reference.

Reply:
discussed. reverting to the original and only showing immutable props. In this case, `data  source` and `url`.